auth/grpctransport/grpctransport.go

```diff
@@ -239,16 +243,20 @@ func dial(ctx context.Context, secure bool, opts *Options) (*grpc.ClientConn, er
 
 // grpcTokenProvider satisfies https://pkg.go.dev/google.golang.org/grpc/credentials#PerRPCCredentials.
 type grpcTokenProvider struct {
-	auth.TokenProvider
+	creds *detect.Credentials // TODO(chrisdsmith): convert usages of TokenProvider to creds
 
 	secure bool
 
 	// Additional metadata attached as headers.
-	metadata map[string]string
+	metadata             map[string]string
+	clientUniverseDomain string
 }
 
 func (tp *grpcTokenProvider) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
-	token, err := tp.Token(ctx)
+	if err := tp.creds.ValidateUniverseDomain(ctx, tp.clientUniverseDomain); err != nil {
```

Comment:
consider making this a top level function in the base internal package

Reply:
Done, thanks!